src/devices/regency_fan.c

```diff
@@ -0,0 +1,230 @@
+    /* 0  */ "invalid",
+    /* 1  */ "fan_speed",
+    /* 2  */ "fan_speed",
+    /* 3  */ "invalid",
+    /* 4  */ "light_intensity",
+    /* 5  */ "light_delay",
+    /* 6  */ "fan_direction",
+    /* 7  */ "invalid",
+    /* 8  */ "invalid",
+    /* 9  */ "invalid",
+    /* 10 */ "invalid",
+    /* 11 */ "invalid",
+    /* 12 */ "invalid",
+    /* 13 */ "invalid",
+    /* 14 */ "invalid",
+    /* 15 */ "invalid"
+};
+
+static int regency_fan_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+
+    data_t *data = NULL;
+    int index = 0; // a row index
+    int num_bits = 0;
+    int debug_output = decoder->verbose;
+    int return_code = 0;
+
+    if (debug_output > 1) {
+        bitbuffer_printf(bitbuffer, "%s: ", __func__);
+    }
+
+    if (bitbuffer->num_rows < 1) {
+        if (debug_output > 1) {
+            fprintf(stderr, "No rows.\n");
+        }
+
+        return 0;
+    }
+
+    bitbuffer_invert(bitbuffer);
+
+    for (index = 0; index < bitbuffer->num_rows; index++) {
+        num_bits = bitbuffer->bits_per_row[index];
+
+        if (num_bits != NUM_BITS) {
+            if (debug_output > 1) {
+                fprintf(stderr, "Expected %d bits, got %d.\n", NUM_BITS, num_bits);
+            }
+
+            continue;
+        }
+
+        uint8_t bytes[NUM_BYTES];
+        bitbuffer_extract_bytes(bitbuffer, index, BYTE_START, bytes, NUM_BITS);
+        reflect_bytes(bytes, NUM_BYTES);
+
+        // Calculate nibble sum and compare
+        int checksum = add_nibbles(bytes, 2) & 0x0f;
+        if (checksum != bytes[SUM_BYTE]) {
+            if (debug_output > 1) {
+                fprintf(stderr, "Checksum failure: expected %0x, got %0x\n", bytes[SUM_BYTE], checksum);
+            }
+
+            continue;
+        }
+
+        /*
+         * Now that message "envelope" has been validated, start parsing data.
+         */
+        uint8_t command = bytes[CMD_CHAN_BYTE] >> 4;
+        uint8_t channel = ~bytes[CMD_CHAN_BYTE] & 0x0f;
+        uint32_t value = bytes[VALUE_BYTE];
+        char value_string[64] = {0};
+
+        switch(command) {
+            case CMD_STOP:
+                sprintf(value_string, "stop");
+                break;
+
+            case CMD_FAN_SPEED:
+                sprintf(value_string, "speed %d", value);
+                break;
+
+            case CMD_LIGHT_INT:
+                sprintf(value_string, "%d %%", value);
+                break;
+
+            case CMD_LIGHT_DELAY:
+                sprintf(value_string, "%s", value == 0 ? "off" : "on");
+                break;
+
+            case CMD_FAN_DIR:
+                sprintf(value_string, "%s", value == 0x07 ? "clockwise" : "counter-clockwise");
+                break;
+
+            default:
+                if (debug_output > 1) {
+                    fprintf(stderr, "Unknown command: %d\n", command);
+                    continue;
+                }
+                break;
+        }
+
+        return_code = 1;
```

Comment:
`        return_code += 1;`

Reply:
Why would you add the possibility of returning a non- 0 or 1 value here? If the return_code is (erroneously) set in the code in a leter edit, you run the risk of returning something other than 0 or 1.

Reply:
return is the *number of* events generated, or an error code.